### PR TITLE
Example finding coauthored papers b/w author pair

### PR DESCRIPTION
**Pull request by dirkraft**
https://github.com/allenai/scholar/issues/36322

~ultralite test script included for periodically rerunning the script~ removed, we don't want to litter our examples with repeat environment setup code.

test script included. you can see output here https://github.com/allenai/s2-folks/actions/runs/4995026522/jobs/8946359836#step:4:1

### CODE REVIEW

**Comment by dirkraft** [comment]
Now using the super basic CI for python examples. Here's the output of this example's `test` https://github.com/allenai/s2-folks/actions/runs/4995026522/jobs/8946359836

File: examples/python/find_coauthored_papers/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Given two author names, this script finds coauthored papers.
+In the case that multiple authors match either author name,
```
Comment (AmberRose2) [inline examples/python/find_coauthored_papers/README.md:2]: Nit: this wording is a little confusing to me, I find something like "In the case there are multiple matches for either author's name" a bit more clear.

File: examples/python/find_coauthored_papers/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+a summary of each author is printed. The user should then re-run
```
Comment (AmberRose2) [inline examples/python/find_coauthored_papers/README.md:3]: Nit: "a summary of each match is printed" just feeling like we use the word author to refer to the user input already so confusing to use it again.

**Review by AmberRose2 — APPROVED** [review]
Little nits with wording, but LGTM!

**Comment by wammar** [comment]
So excited to have our first example out!! I'm thinking about adding one line for each example contributed to the repo in the release notes, in addition to sharing them on the slack channel. @dirkraft  Please let me know if you have any suggestions/thoughts on sharing this with the developer community around our APIs.

**Pull request merged**